games/pacman.puml

```diff
@@ -0,0 +1,186 @@
+@startuml
+package "Process Input Handler" {
+!includesub ../uml/frontend-software-system/process-input-handler.puml!INTERFACE
+}
+package "Render Handler" {
+!includesub ../uml/frontend-software-system/render-handler.puml!INTERFACE
+}
+!include ../uml/frontend-software-system/game-settings.puml
+
+
+package Pac-Man {
+
+class Position {
+ - x : int
+ - y : int
+}
+
+class Wall {
+ - sprite : Sprite
+ - position: Position
+}
+
+class PowerSphere
+{
+ - sprite : Sprite
+ - time : int
+ - position: Position
+ + collect() : int
+ + activatePower() : void
+}
+
+class Sphere
+{
+ - sprite : Sprite
+ - position: Position
+ + collect() : int
+}
+
+abstract class Ghost
+{
+ - sprite : Sprite
+ -state : string
+ -color : string
+ + fleePacman () : void
+ + chasePacman () : void
+}
+
+class ClydeGhost{}
+class PinkyGhost{}
+class InkyGhost{}
+class BlinkyGhost{}
+
+interface ICharacter
+{
+ - position: Position
+ - direction : Direction
+ + respawn (Position position) : void
+ + move (Position) : bool
+}
+class Pacman 
+{
+ - sprite : Sprite
+ - lives : int
```

Comment:
how do you know if pacman is still alive?

Reply:
if lives equals 0, but a new isAlive method was added in Pacman class.